test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -383,8 +392,92 @@ private void verifyNoData(Map<String, Table<StringKey, IntValue, Metadata>> tabl
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        try (TxnContext txn = srcCorfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CorfuStoreEntry<LogReplicationMetadata.ReplicationStatusKey, LogReplicationMetadata.ReplicationStatusVal, Message>
+                    entry = txn.getRecord(replicationStatusTable, key);
+            if (entry.getPayload() != null) {
+                snapshotSyncStatus = entry.getPayload().getSnapshotSyncInfo().getStatus();
+            }
+            txn.commit();
+        }
+
+        Assert.assertEquals(expectedStatus, snapshotSyncStatus);
+    }
+
+    private void verifyCurrentSyncStatus(LogReplicationMetadata.SyncStatus expectedStatus) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+
+        LogReplicationMetadata.ReplicationStatusKey key = LogReplicationMetadata.ReplicationStatusKey.newBuilder().setClusterId(REMOTE_CLUSTER_ID).build();
+        LogReplicationMetadata.SyncStatus syncStatus = null;
+        Table<LogReplicationMetadata.ReplicationStatusKey, LogReplicationMetadata.ReplicationStatusVal, Message> replicationStatusTable;
+        replicationStatusTable = srcCorfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                REPLICATION_STATUS_TABLE,
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        try (TxnContext txn = srcCorfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CorfuStoreEntry<LogReplicationMetadata.ReplicationStatusKey, LogReplicationMetadata.ReplicationStatusVal, Message>
+                    entry = txn.getRecord(replicationStatusTable, key);
+            if (entry.getPayload() != null) {
+                syncStatus = entry.getPayload().getStatus();
+            }
+            txn.commit();
+        }
+
+        Assert.assertEquals(expectedStatus, syncStatus);
+    }
+
     /* ***************************** LOG REPLICATION IT TESTS ***************************** */
 
+    /**
+     * This test helps to emulate the raceCondition where statusSyncUpdater thread gets precedence over the state transition.
+     * A statusSyncUpdater service is run independently without the FSM and
+     * NOT_STARTED status is verified from the replicationStatus table.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testSyncStatusBeforeSyncState() throws Exception {
+        setupEnv();
+
+        LogReplicationMetadataManager srcManager = new LogReplicationMetadataManager(srcTestRuntime, 0, ACTIVE_CLUSTER_ID);
+        LogReplicationConfigManager tableManagerPlugin = new LogReplicationConfigManager(srcTestRuntime);
```

Comment:
it would nice to have this as configManger, because the LRConfigManager discovers streamsToReplicate and also compares versions during upgrade scenarios.

tableManagerPlugin becomes difficult to interpret without the type

Reply:
Got it. Updated likewise.